### PR TITLE
[FrontendProxy] Add tracing to Envoy

### PR DESCRIPTION
**Pull request by julianocosta89**
# Changes

As Envoy supports OpenTelemetry, I believe it would be great if we could use it.
https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/opentelemetry

This PR adds the configuration required to add tracing to Envoy.

jaeger-proxy

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [x] Appropriate documentation updates in the [docs](../docs/) folder

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

### CODE REVIEW

**Comment by cartersocha** [comment]
We should add envoy to the current services architecture doc. Especially if there's a span coming from it

Should we add a simple envoy service doc with config instructions too?

**Comment by julianocosta89** [comment]
@cartersocha Not sure what exactly to write, but I've added the "how to" make Envoy produce spans.

**Comment by cartersocha** [comment]
I'm seeing errors on the frontend and checkout service

**Comment by cartersocha** [comment]
Not sure why the shipping service is throwing issues. 

"13 INTERNAL: shipping quote failure: failed to get shipping quote: rpc error: code = Unknown desc = error sending request for url (http://quoteservice:8090/getquote): error trying to connect: dns error: failed to lookup address information: Name does not resolve"
--

**Comment by cartersocha** [comment]
A system prune cleared up the errors but I'm not seeing an envoy span in any trace. I see envoy as an orphan span currently and the main trace doesn't have it present

**Comment by julianocosta89** [comment]
@cartersocha true!
Something that I forgot to mention is that this is generating spans only when you navigate to the OTel-Demo page.
The loadgen doesn't send requests through the envoy, but to the frontend container directly.

**Comment by cartersocha** [comment]
I see! lgtm. 

This is where it would be nice to have non synthetic requests set the attribute to filter on.

**Review by cartersocha — APPROVED** [review]
Please add a note in the envoy services doc at the top saying only non synthetic requests will trigger the envoy tracing. 

Lgtm otherwise

File: docker-compose.yml [hunk 1]
```diff
@@ -262,6 +262,8 @@ services:
       - GRAFANA_SERVICE_HOST
       - JAEGER_SERVICE_PORT
       - JAEGER_SERVICE_HOST
+      - OTEL_COLLECTOR_HOST
```
Comment (puckpuck) [inline docker-compose.yml:265]: I think these environment variables belong in the `# Collector` section instead.  Though only used by Envoy, we should keep these vars with the other similar ones.

**Review by puckpuck — APPROVED** [review]
One small nit. everything works as expected. 👍

File: docker-compose.yml (hunk 1 above)
Comment (julianocosta89) [reply docker-compose.yml:329]: @puckpuck I think you meant in the `.env` file, right?

Comment (puckpuck) [reply docker-compose.yml:329]: yes I meant in the .env file

**Pull request merged**